src/ui/hooks/useDebounce.ts

```diff
@@ -0,0 +1,33 @@
+import { useCallback, useRef } from "react";
+
+interface DebounceOptions {
+  wait?: number;
+  immediate?: boolean;
+}
+
+export function useDebounce<T extends any[]>(
+  callback: (...args: T) => void,
+  options?: DebounceOptions
+): (...args: T) => void {
+  const timeoutIdRef = useRef<NodeJS.Timeout | null>(null);
```

Comment:
Instead of `NodeJS.Timeout` can you use `number`. In order to make that type match up with the browser's `setTimeout` method, you'll need to use `window.setTimeout()` which has a return type of `number`.